src/docfx/build/legacy/LegacyDependencyMap.cs

```diff
@@ -17,7 +15,7 @@ internal static class LegacyDependencyMap
         {
             using (Progress.Start("Convert Legacy Dependency Map"))
             {
-                var legacyDependencyMap = new ConcurrentBag<LegacyDependencyMapItem>();
+                var legacyDependencyMap = new List<LegacyDependencyMapItem>();
```

Comment:
better yet, we can have an `ArrayBuilder` and a `DictionaryBuilder` that encapsulate the lock logic?

Reply:
sounds good